emergence_lib/src/simulation/mod.rs

```diff
@@ -27,18 +27,29 @@ impl Plugin for SimulationPlugin {
             .add_plugin(StructuresPlugin)
             .add_plugin(UnitsPlugin)
             .add_plugin(SignalsPlugin)
-            .add_system_to_stage(CoreStage::PreUpdate, create_passable_filter);
+            .add_startup_system_to_stage(StartupStage::PostStartup, initialize_passable_filter)
+            .add_system_to_stage(CoreStage::PreUpdate, update_passable_filter);
     }
 }
 
-/// Create the [`PassableFilters`] resource
-pub fn create_passable_filter(
-    mut commands: Commands,
-    impassable_query: Query<&TilePos, With<Impassable>>,
-    map_positions: Res<MapPositions>,
+/// Create the [`PassabilityCache`] resource
+pub fn initialize_passable_filter(mut commands: Commands, map_positions: Res<MapPositions>) {
+    commands.insert_resource(PassabilityCache::new(&map_positions));
+}
+
+// /// Create the [`PassableFilters`] resource
+// pub fn update_passable_filter(
+//     newly_impassable: Query<&TilePos, (With<Impassable>, Changed<Impassable>)>,
+//     newly_passable: Query<&TilePos, (Without<Impassable>, Changed<Impassable>)>,
+//     mut passable_filters: ResMut<PassableFilters>,
+// ) {
+//     passable_filters.update_from_changed_passable_queries(&newly_impassable, &newly_passable);
+// }
```

Comment:
We can probably remove this if it's not needed anymore

Reply:
@TimJentzsch not totally sure if it's needed. This one is slightly different in that it uses a `Changed` filter on top of the `With/Without<Impassable>` but yeah, I can remove it for now